### PR TITLE
[ADD]mrp: add product category on bill of bill of materials

### PR DESCRIPTION
**Pull request by gan-odoo**
Task : https://www.odoo.com/web#id=11392&view_type=form&model=project.task&menu_id=3942&action=327
Pad : https://pad.odoo.com/p/r.cUcr7narfbRhnYOe

### CODE REVIEW

File: addons/mrp/mrp.py [hunk 1]
```diff
@@ -155,6 +155,12 @@ class mrp_bom(osv.osv):
     _description = 'Bill of Material'
     _inherit = ['mail.thread']
 
+    def get_product_categ_ids(self, cr, uid, ids, context=None):
```
Comment (ara-odoo) [inline addons/mrp/mrp.py]: I think name should be get_bom_ids??

Comment (gan-odoo) [inline addons/mrp/mrp.py]: Ok Done!

File: addons/mrp/mrp.py [hunk 2]
```diff
@@ -155,6 +155,12 @@ class mrp_bom(osv.osv):
     _description = 'Bill of Material'
     _inherit = ['mail.thread']
 
+    def _get_bom_ids(self, cr, uid, ids, context=None):
+        result = {}
+        for bom in self.pool.get('product.template').browse(cr, uid, ids, context=context).bom_ids:
+            result[bom.id] = True
+        return result.keys()
```
Comment (rim-odoo) [inline addons/mrp/mrp.py]: Why building a dictionary for returning only its keys? Why not a list of bom.id's?

File: addons/mrp/mrp.py [hunk 3]
```diff
@@ -155,6 +155,12 @@ class mrp_bom(osv.osv):
     _description = 'Bill of Material'
     _inherit = ['mail.thread']
 
+    def _get_bom_ids(self, cr, uid, ids, context=None):
+        result = {}
+        for bom in self.pool.get('product.template').browse(cr, uid, ids, context=context).bom_ids:
```
Comment (rim-odoo) [inline addons/mrp/mrp.py]: `self.pool.['product.template']`

Comment (rim-odoo) [inline addons/mrp/mrp.py]: What will happen if browse returns more than one browse record? We take only the bom_ids of the first one? That's probably incorrect

File: addons/mrp/mrp.py [hunk 4]
```diff
@@ -167,6 +173,11 @@ class mrp_bom(osv.osv):
             domain="['&', ('product_tmpl_id','=',product_tmpl_id), ('type','!=', 'service')]",
             help="If a product variant is defined the BOM is available only for this product."),
         'bom_line_ids': fields.one2many('mrp.bom.line', 'bom_id', 'BoM Lines', copy=True),
+        'categ_id': fields.related('product_tmpl_id', 'categ_id', type='many2one', relation='product.category', string='Product Category', readonly=True,
+            store = {
+                'mrp.bom': (lambda self, cr, uid, ids, c={}: ids, ['product_tmpl_id'], 10),
```
Comment (rim-odoo) [inline addons/mrp/mrp.py]: c={} is incorrect, but has no impact here. Always do c=None in function signature

**Pull request closed without merging**